src/json-editor.component.scss

```diff
@@ -184,10 +195,105 @@ body, html {
   }
 
   .menu-container {
-    background-color: #1D2D3D;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    background-color: #2c3e50;
     height: 100%;
-    overflow: auto;
+    width: 0%;
+    overflow-x: hidden;
+    opacity: 0;
+    visibility: hidden;
     @include top-level-add-field-dropdown-container;
+    transition: all 0.3s ease-in;
+    #close-sidebar {
+      cursor: pointer;
+      width: 100%;
+      display: flex;
+      margin-top: 20px;
+      color: white;
+      i {
+        margin-left: 10px;
+      }
+    }
+    &.open {
+      opacity: 1;
+      visibility: visible;
+      width: 12%;
+      padding-left: 5px;
+    }
+  }
+
+  .collapsed-menu-container {
+    position: relative;
+    background-color: #1D2D3D;
+    width: $default-sidebar-width;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-end;
+    align-items: center;
+    .add-field-button {
+      position: absolute;
+      top: 0px;
+      margin-top: 10px;
+      display: flex;
+      justify-content: center;
+      width: 100%;
+      cursor: pointer;
+      color: #ddd;
+      i {
+        transition: all 0.5s ease-in-out;
+        -webkit-transition: all 0.5s ease-in-out;
+        -ms-transition: all 0.5s ease-in-out;
+        &:hover {
+          color: white;
+          text-shadow: 3px 3px 14px #2a5d88;
+        }
+      }
+    }
+    #open-sidebar {
```

Comment:
using `id` to apply styles only acceptable if you have to do really hacky thing which is discouraged :).

if you need to this in a `e2e` test, you might use `id` but write another class to appy the styles.

(`#ng2-json-editor` is an exception, since it's root selector used to scope our global styles)